go/vt/sqlparser/sql.y

```diff
@@ -602,19 +615,56 @@ base_select:
   }
 
 base_select_no_cte:
-  SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list where_expression_opt group_by_opt having_opt window_opt
+  SELECT comment_opt cache_opt distinct_opt sql_calc_found_rows_opt straight_join_opt select_expression_list into_opt from_opt where_expression_opt group_by_opt having_opt window_opt into_opt
   {
-    $$ = &Select{Comments: Comments($2), Cache: $3, Distinct: $4, Hints: $6, SelectExprs: $7, From: TableExprs{&AliasedTableExpr{Expr:TableName{Name: NewTableIdent("dual")}}}, Where: NewWhere(WhereStr, $8), GroupBy: GroupBy($9), Having: NewWhere(HavingStr, $10), Window: $11}
+    if $8 == nil {
+      $8 = $14
+    } else if $14 != nil {
+      yylex.Error(fmt.Errorf("Multiple INTO clauses in one query block").Error())
```

Comment:
It would be nice to add a test for this error throwing logic – that helps ensure we got the code right and also that this behavior won't regress in the future.